File: app/src/main/java/com/doubean/ford/data/vo/Group.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.time.LocalDateTime;
import java.util.List;

@Entity(tableName = "groups")
public class Group {

    @PrimaryKey
    @NonNull
    public String id;

    @NonNull
    public String name;

    @SerializedName("member_count")
    public Integer memberCount;

    @SerializedName("topic_count")
    public Integer postCount;

    @SerializedName("create_time")
    public LocalDateTime dateCreated;

    @SerializedName("sharing_url")
    public String url;

    public String uri;

    @SerializedName("avatar")
    public String avatarUrl;

    @SerializedName("member_name")
    public String memberName;

    @SerializedName("desc_abstract")
    public String shortDescription;

    @SerializedName("desc")
    public String description;

    @SerializedName("group_tabs")
    public List<GroupTab> tabs;

    @SerializedName("post_tags_normal")
    public List<GroupPostTag> postTags;

    @SerializedName("background_mask_color")
    public String colorString;

    public Group() {
    }


}

File: app/src/main/java/com/doubean/ford/data/vo/Group.kt
package com.doubean.ford.data.vo

import androidx.room.*
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

@Entity(tableName = "groups")
class Group (
    @PrimaryKey
    val id: String,
    val name: String,

    @SerializedName("member_count")
    val memberCount: Int? = null,

    @SerializedName("topic_count")
    val postCount: Int? = null,

    @SerializedName("create_time")
    val dateCreated: LocalDateTime? = null,

    @SerializedName("sharing_url")
    val url: String,
    val uri: String,

    @SerializedName("avatar")
    val avatarUrl: String? = null,

    @SerializedName("member_name")
    val memberName: String? = null,

    @SerializedName("desc_abstract")
    val shortDescription: String? = null,

    @SerializedName("desc")
    val description: String? = null,

    @SerializedName("group_tabs")
    val tabs: List<GroupTab>? = null,

    @SerializedName("post_tags_normal")
    val postTags: List<GroupPostTag>? = null,

    @SerializedName("background_mask_color")
    val colorString: String? = null
)